lib/Echidna/ABI.hs

```diff
@@ -155,3 +161,68 @@ displayAbiCall (t, vs) = unpack t ++ "(" ++ L.intercalate "," (map prettyPrint v
 -- the form (Function name, [arg0 type, arg1 type...])
 genInteractions :: MonadGen m => [SolSignature] -> m SolCall
 genInteractions ls = genAbiCall =<< Gen.element ls
+
+type Listy t a = (IsList (t a), Item (t a) ~ a)
+
+switchElem :: (Listy t a, MonadGen m) => m a -> t a -> m (t a)
+switchElem g t = let l = toList t; n = length l in do
+  i <- Gen.element [0..n]
+  x <- g
+  return . fromList $ take i l <> [x] <> drop (i+1) l
+
+changeChar :: MonadGen m => ByteString -> m ByteString
+changeChar = fmap BS.pack . switchElem Gen.enumBounded . BS.unpack
+
+addBS :: MonadGen m => ByteString -> m ByteString
+addBS b = Gen.element [(<> b), (b <>)] <*> Gen.utf8 (constant 0 (256 - BS.length b)) Gen.unicode
+
+dropBS :: MonadGen m => ByteString -> m ByteString
+dropBS b = Gen.choice [ BS.drop <$> Gen.element [1..BS.length b]   <*> pure b
+                      , BS.take <$> Gen.element [0..BS.length b-1] <*> pure b
+                      ]
+
+changeBS :: MonadGen m => ByteString -> m ByteString
+changeBS b = Gen.choice $ [changeChar, addBS, dropBS] <&> ($ b)
+
+changeNumber :: (Enum a, Num a, MonadGen m) => a -> m a
```

Comment:
Does this work for uint and address? You may wind up unintentionally mutating the input by a huge amount if there's an underflow.